### ISSUE TITLE
Examples: Where is err being defined in the Running tasks in series example?

### ISSUE THREAD

**Issue by JamesMcMahon** [gulpjs/gulp#1092]
In the [Running tasks in series, i.e. Task Dependency](https://github.com/gulpjs/gulp/blob/master/docs/recipes/running-tasks-in-series.md) example, where is `err` being defined in this code block?

```js
var gulp = require('gulp');

// takes in a callback so the engine knows when it'll be done
gulp.task('one', function(cb) {
    // do stuff -- async or otherwise
    cb(err); // if err is not null and not undefined, the orchestration will stop, and 'two' will not run
});

// identifies a dependent task must be complete before this one begins
gulp.task('two', ['one'], function() {
    // task 'one' is done now
});

gulp.task('default', ['one', 'two']);
// alternatively: gulp.task('default', ['two']);
```

I'm new to Gulp, so please excuse me if I am missing something obvious, but looking at the example it's not clear to me.

**Comment by jmm** [comment]
If you perform some operation in the task that may result in an error you could pass that to the callback, e.g.:

```js
try {
  something();
}
catch (e) { cb(e); }
```
Or if you perform some async operation upon which success or failure of the task hinges and that takes a standard node-style callback, you can just pass the callback to that operation. E.g.:

```js
gulp.task('one', function (cb) {
    fs.writeFile('filename', 'data', cb);
});
```

**Comment by JamesMcMahon** [comment]
That makes sense. The example just omits that in kind of a strange way.

When I get a chance I'll put a suggested change in a PR. Maybe something like:

```js
// takes in a callback so the engine knows when it'll be done
gulp.task('one', function(cb) {
    try {
        // do stuff -- async or otherwise
    } catch (err) {
        cb(err); // if err is not null and not undefined, the orchestration will stop, and 'two' will not run
   }
});
```

**Comment by phated** [comment]
@JamesMcMahon we'd be down for a PR but try-catch doesn't work in an async context so the example you gave doesn't make complete sense

**Comment by JamesMcMahon** [comment]
Sorry for the late response, I've been busy.

Does `err` ever make sense in an async context? Would it be better to just remove `err` from the examples?

**Comment by jmm** [comment]
@JamesMcMahon 
> Does `err` ever make sense in an async context?

Yes, usually.

```js
gulp.task('one', function (cb) {
  fs.writeFile('filename', 'data', opts, function (err) {
    if (err) return cb(err);
    console.log("Success!");
    cb(null);
  });
});
```

**Comment by phated** [comment]
Closed by #1169
